Problem should rather be fixed in jpeglib.h on Solaris where
jpeglib.h do not behave as "system library header" from the
C++ point of view because the system headers should define
C++ compatibility internally.

Not suitable for community.

--- a/include/qpdf/Pl_DCT.hh	2018-01-11 07:11:29.870607412 +0000
+++ b/include/qpdf/Pl_DCT.hh	2018-01-11 07:15:35.287505262 +0000
@@ -10,7 +10,9 @@
 
 #include <qpdf/Pipeline.hh>
 #include <qpdf/Pl_Buffer.hh>
+extern "C" {
 #include <jpeglib.h>
+}
 
 class Pl_DCT: public Pipeline
 {
